allennlp/training/trainer.py

```diff
@@ -926,13 +939,11 @@ def from_params(cls,
         patience = params.pop_int("patience", None)
         validation_metric = params.pop("validation_metric", "-loss")
         num_epochs = params.pop_int("num_epochs", 20)
-        cuda_device = params.pop_int("cuda_device", -1)
+        cuda_device = params.pop( "cuda_device")
         grad_norm = params.pop_float("grad_norm", None)
         grad_clipping = params.pop_float("grad_clipping", None)
         lr_scheduler_params = params.pop("learning_rate_scheduler", None)
 
-        if cuda_device >= 0:
```

Comment:
If the model is always none now then maybe it should be removed as a parameter from the trainer constructor?

Reply:
That wasn't in _my_ changes though, right? Pretty sure upstream modified the `trainer.py` quite a bit so you may want to diff against their master first.